### PR TITLE
[BUGFIX beta] Set models array length in link-to.

### PR DESCRIPTION
**Pull request by nathanhammond**
We know the array length in advance so we should set it.

Proof:
http://jsperf.com/array-push-new-literal

### CODE REVIEW

**Comment by stefanpenner** [comment]
LGTM

File: packages/ember-routing-views/lib/components/link-to.js [hunk 1]
```diff
@@ -769,7 +769,7 @@ let LinkComponent = EmberComponent.extend({
     this.set('queryParams', queryParams);
 
     // 4. Any remaining indices (excepting `targetRouteName` at 0) are `models`.
-    let models = [];
+    let models = new Array(params.length - 1);
```
Comment (rwjblue) [inline packages/ember-routing-views/lib/components/link-to.js]: This seems to be throwing an error when `params` is empty.

screenshot

Comment (stefanpenner) [inline packages/ember-routing-views/lib/components/link-to.js]: if params are empty, we may want to skip allocating at all.

Comment (rwjblue) [inline packages/ember-routing-views/lib/components/link-to.js]: ya, agreed

**Comment by rwjblue** [comment]
Once that last issue is fixed, can you prefix with `[BUGFIX beta]`?  Seems pretty safe to pull down into beta, and we'll take all the perfs we can get!

**Comment by stefanpenner** [comment]
It would be nice if provided perfs had demonstrable improvements in ember. An example or benchmark illustrating this improvement would be nice.

**Comment by krisselden** [comment]
@stefanpenner agreed, often micro benchmarks like this have no noticeable impact on a macro benchmark due to the improved code being a small percentage of the macro case.  I'm fine with merging but am curious if this has any impact on a benchmark of list with link-to components.

**Comment by nathanhammond** [comment]
In emberperf it is about 2% faster across multiple runs, presumably by just pulling it out into a function which can be optimized separately, but that doesn't exercise the models use case where we should actually get our performance win.

Tests pass now.

### Ember Version: link-to modifications

```
.---------------------------------------------------------------.
|               Ember Performance Suite - Results               |
|---------------------------------------------------------------|
|           Name           | Speed  | Error  | Samples |  Mean  |
|--------------------------|--------|--------|---------|--------|
| Render List with link-to |   7.23 | 403.89 |      91 | 138.26 |
| Render link-to           | 379.13 |   5.93 |    1010 |   2.64 |
'---------------------------------------------------------------'
```

### Ember Version: Canary

```
.---------------------------------------------------------------.
|               Ember Performance Suite - Results               |
|---------------------------------------------------------------|
|           Name           | Speed  | Error  | Samples |  Mean  |
|--------------------------|--------|--------|---------|--------|
| Render List with link-to |   6.87 | 528.64 |      86 | 145.51 |
| Render link-to           | 371.58 |   4.84 |    1004 |   2.69 |
'---------------------------------------------------------------'
```

**Comment by mixonic** [comment]
Thanks @nathanhammond :-)

**Pull request merged**